pkg/products/marin3r/reconciler.go

```diff
@@ -263,6 +263,12 @@ func (r *Reconciler) preUpgradeBackupExecutor() backup.BackupExecutor {
 	if r.installation.Spec.UseClusterStorage != "false" {
 		return backup.NewNoopBackupExecutor()
 	}
-	//todo add backup for redis once it's added to the reconciler
-	return backup.NewNoopBackupExecutor()
+
+	return backup.NewConcurrentBackupExecutor(
+		backup.NewAWSBackupExecutor(
+			r.installation.Namespace,
+			"ratelimit-service-redis-rhmi",
+			backup.RedisSnapshotType,
+		),
+	)
```

Comment:
```suggestion
	return backup.NewAWSBackupExecutor(
			r.installation.Namespace,
			"ratelimit-service-redis-rhmi",
			backup.RedisSnapshotType,
	)
```

I would only use the `ConcurrentBackupExecutor` for one or more resources that need backup. Otherwise we'll create a single new goroutine and wait for it, which has no benefit

Reply:
Good call